### PR TITLE
Jc/ft6xx6x samples

### PR DESCRIPTION
**Pull request by jcabad100**
M5Stack Core2 Touch support refactored (5 new files + new Program.cs ) 

## Description
All finding from previous canceled PR implemeted.

## Motivation and Context
M5Stack Core2 Touch support to discriminate 3 (virtual) buttons and/or one point touch on the screen.

## How Has This Been Tested?
- VS 2022 Community Edition on Windows 10
- M5Stack Core2 target

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by josesimoes** [comment]
@jcabad100 this new class is not being used in the sample...

**Comment by jcabad100** [comment]
> @jcabad100 this new class is not being used in the sample...

Jose, of couse all the new classes are used in the NEW version of the pre-existing sample Program.cs which was PR separately ...i ddi not know how to do that in the same PR....

**Comment by jcabad100** [comment]
> @jcabad100 this new class is not being used in the sample...

I just re-do the PR using the same branch as i used for all the other 5 new files/classes ...please check 
https://github.com/jcabad100/nanoFramework.IoT.Device/blob/JC/Ft6xx6x-samples/devices/Ft6xx6x/samples/Program.cs

**Comment by josesimoes** [comment]
> Jose, of couse all the new classes are used in the NEW version of the pre-existing sample Program.cs which was PR separately ...i ddi not know how to do that in the same PR....

Seems that _now_ the sample is using it!

**Comment by jcabad100** [comment]
> 

in fact now the automatic build reports success too..

File: devices/Ft6xx6x/samples/Program.cs [hunk 1]
```diff
@@ -1,55 +1,83 @@
...
+                    TouchPanelDiscriminator.DiscrimitateButtons(point.X, point.Y);
+                }
+                else if (points == 2)
+                {
+                    var dp = sensor.GetDoublePoints();
+                    Debug.WriteLine($"ID: {dp.Point1.TouchId}, X: {dp.Point1.X}, Y: {dp.Point1.Y}, Weight: {dp.Point1.Weigth}, Misc: {dp.Point1.Miscelaneous}, Event: {dp.Point1.Event}");
+                    Debug.WriteLine($"ID: {dp.Point2.TouchId}, X: {dp.Point2.X}, Y: {dp.Point2.Y}, Weight: {dp.Point2.Weigth}, Misc: {dp.Point2.Miscelaneous}, Event: {dp.Point1.Event}");
+                }
+            }
+
```
Comment (Ellerbach) [inline devices/Ft6xx6x/samples/Program.cs:76]: 2 extra lines you should remove

File: devices/Ft6xx6x/samples/TouchEventArgs.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    public class TouchEventArgs : EventArgs
+    {
+        public EventCategory TouchEventCategory { get; set; }
+        public int TouchEventSubCategory { get; set; }
+        public int X1 { get; set; }
+        public int Y1 { get; set; }
+        public int X2 { get; set; }
+        public int Y2 { get; set; }
+
+        //NOTE: In order to be able to differentiate a short press from a long press a time stamp maybe needed later
```
Comment (Ellerbach) [inline devices/Ft6xx6x/samples/TouchEventArgs.cs:15]: for all comments, please have a space after the // so it does improve the readability.

File: devices/Ft6xx6x/samples/TouchEventSubcategory.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿namespace NFAppCore2App2
```
Comment (Ellerbach) [inline devices/Ft6xx6x/samples/TouchEventSubcategory.cs:1]: You are missing the header here as well

File: devices/Ft6xx6x/samples/TouchPanelDiscriminator.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
+﻿using System;
```
Comment (Ellerbach) [inline devices/Ft6xx6x/samples/TouchPanelDiscriminator.cs:1]: And header here as well :-)

File: devices/Ft6xx6x/samples/TouchPanelDiscriminator.cs [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            {
+                touchEventArgs.TouchEventSubCategory = (int)TouchEventSubcategory.RightButton;
+            }
+            else
+            {
+                touchEventArgs.TouchEventSubCategory = (int)TouchEventSubcategory.SingleTouch;
+            }
+
+            TouchPanelProcessor.OnTouchedEvent(touchEventArgs.TouchEventSubCategory, touchEventArgs.X1, touchEventArgs.Y1);
+
```
Comment (Ellerbach) [inline devices/Ft6xx6x/samples/TouchPanelDiscriminator.cs:58]: extra line here

File: devices/Ft6xx6x/samples/TouchPanelProcessor.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+            args.TouchEventSubCategory = eventSubCategory;
+            args.X1 = x1;
+            args.Y1 = y1;
+            args.X2 = x2;
+            args.Y2 = y2;
+
+            TouchedEvent?.Invoke(null, args);
+
+        }
+
```
Comment (Ellerbach) [inline devices/Ft6xx6x/samples/TouchPanelProcessor.cs:25]: extra line here

File: devices/Ft6xx6x/samples/TouchPanelProcessor.cs [hunk 7]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
```
Comment (Ellerbach) [inline devices/Ft6xx6x/samples/TouchPanelProcessor.cs:1]: Missing headers as well here

File: devices/Ft6xx6x/samples/TouchPanelUserAppLogic.cs [hunk 8]
```diff
@@ -0,0 +1,58 @@
+﻿using System;
```
Comment (Ellerbach) [inline devices/Ft6xx6x/samples/TouchPanelUserAppLogic.cs:1]: Missing headers here

File: devices/Ft6xx6x/samples/TouchPanelUserAppLogic.cs [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+            else if (e.TouchEventSubCategory == (int)TouchEventSubcategory.DoubleTouch)
+            {
+                Debug.WriteLine(strXY1XY2);
+
+            }
+            else
+            {
+                Debug.WriteLine("ERROR: UKNOWN TouchEventSubcategory");
+            }
+
```
Comment (Ellerbach) [inline devices/Ft6xx6x/samples/TouchPanelUserAppLogic.cs:54]: extra line here

File: devices/Ft6xx6x/samples/TouchPanelUserAppLogic.cs [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+        }
+
```
Comment (Ellerbach) [inline devices/Ft6xx6x/samples/TouchPanelUserAppLogic.cs:56]: extra line here

File: devices/Ft6xx6x/samples/TouchPanelUserAppLogic.cs [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+            else if (e.TouchEventSubCategory == (int)TouchEventSubcategory.SingleTouch)
+            {
+                Debug.WriteLine(strXY1 + e.X1.ToString() + strXY2 + e.Y1.ToString());
+                nanoFramework.Console.WriteLine(strXY1 + e.X1.ToString() + strXY2 + e.Y1.ToString());
+
+            }
+            else if (e.TouchEventSubCategory == (int)TouchEventSubcategory.DoubleTouch)
+            {
+                Debug.WriteLine(strXY1XY2);
+
```
Comment (Ellerbach) [inline devices/Ft6xx6x/samples/TouchPanelUserAppLogic.cs:48]: extra line here

File: devices/Ft6xx6x/samples/TouchPanelUserAppLogic.cs [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+            {
+                Debug.WriteLine(strRB);
+                nanoFramework.Console.WriteLine(strRB);
+
+            }
+            else if (e.TouchEventSubCategory == (int)TouchEventSubcategory.SingleTouch)
+            {
+                Debug.WriteLine(strXY1 + e.X1.ToString() + strXY2 + e.Y1.ToString());
+                nanoFramework.Console.WriteLine(strXY1 + e.X1.ToString() + strXY2 + e.Y1.ToString());
+
```
Comment (Ellerbach) [inline devices/Ft6xx6x/samples/TouchPanelUserAppLogic.cs:43]: extra line here

File: devices/Ft6xx6x/samples/TouchPanelUserAppLogic.cs [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+            {
+                Debug.WriteLine(strMB);
+                nanoFramework.Console.WriteLine(strMB);
+
+            }
+            else if (e.TouchEventSubCategory == (int)TouchEventSubcategory.RightButton)
+            {
+                Debug.WriteLine(strRB);
+                nanoFramework.Console.WriteLine(strRB);
+
```
Comment (Ellerbach) [inline devices/Ft6xx6x/samples/TouchPanelUserAppLogic.cs:37]: extra line here

File: devices/Ft6xx6x/samples/TouchPanelUserAppLogic.cs [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+            {
+                Debug.WriteLine(strLB);
+                nanoFramework.Console.WriteLine(strLB);
+
+            }
+            else if (e.TouchEventSubCategory == (int)TouchEventSubcategory.MiddleButton)
+            {
+                Debug.WriteLine(strMB);
+                nanoFramework.Console.WriteLine(strMB);
+
```
Comment (Ellerbach) [inline devices/Ft6xx6x/samples/TouchPanelUserAppLogic.cs:31]: extra line here

File: devices/Ft6xx6x/samples/TouchPanelUserAppLogic.cs [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+            Debug.WriteLine("Touch Panel Event Received Category= " + e.TouchEventCategory.ToString() + " Subcategory= " + e.TouchEventSubCategory.ToString());
+            
+            nanoFramework.Console.Clear();
+
+            if (e.TouchEventSubCategory == (int)TouchEventSubcategory.LeftButton)
+            {
+                Debug.WriteLine(strLB);
+                nanoFramework.Console.WriteLine(strLB);
+
```
Comment (Ellerbach) [inline devices/Ft6xx6x/samples/TouchPanelUserAppLogic.cs:25]: extra line here

File: devices/Ft6xx6x/samples/TouchPanelUserAppLogic.cs [hunk 16]
```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Diagnostics;
+
+namespace NFAppCore2App2
+{
+    public class TouchPanelUserAppLogic
+    {
+        public void TouchPanelUserAppLogicSkeleton(object sender, TouchEventArgs e)
```
Comment (Ellerbach) [inline devices/Ft6xx6x/samples/TouchPanelUserAppLogic.cs:8]: why not having this function directly as a static in the main program?

File: devices/Ft6xx6x/samples/TouchPanelProcessor.cs [hunk 17]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using nanoFramework.Runtime.Events;
+
+namespace NFAppCore2App2
+{
+    public delegate void TouchEventHandler(object sender, TouchEventArgs e);
+    
+    public class TouchPanelProcessor
```
Comment (Ellerbach) [inline devices/Ft6xx6x/samples/TouchPanelProcessor.cs:8]: if all is static then the class should be static

Comment (Ellerbach) [reply devices/Ft6xx6x/samples/TouchPanelProcessor.cs:8]: Also, looking at the main code, this class is not needed and you can move all the functions directly to the main program. Including the delegate

File: devices/Ft6xx6x/samples/TouchPanelDiscriminator.cs [hunk 18]
```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using nanoFramework.Runtime.Events;
+
+namespace NFAppCore2App2
+{
+    public class TouchPanelDiscriminator
```
Comment (Ellerbach) [inline devices/Ft6xx6x/samples/TouchPanelDiscriminator.cs:6]: this can be a static function in the main program

**Review by Ellerbach — COMMENTED** [review]
Thanks! I've added couple of comments, quite some just for esthetic (extra lines, headers).
And you can simplify the code by moving almost the classes to your main program.

**Comment by Ellerbach** [comment]
@jcabad100 as discussed on the Discord channel, I've adjusted the sample to simplify all the non necessary classes and adjusted some of the naming constant conventions and the few other elements. I hope all makes sense with the PR feedback we gave.
It's really about making things a bit simpler to read, following pattern not to instantiate classes when not needed.
All the code is your original code. No change at all!

**Comment by Ellerbach** [comment]
@jcabad100, I will close this PR. there has been  a lot of changes in the repo and also addition of StyleCop. So Let's close this one and we can always reopen later on.

**Pull request closed without merging**